integration-tests/src/tests/client/features/zero_balance_account.rs

```diff
@@ -0,0 +1,249 @@
+        .build();
+    let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
+
+    let new_account_id: AccountId = "hello.test0".parse().unwrap();
+    let signer0 = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+    let new_signer =
+        InMemorySigner::from_seed(new_account_id.clone(), KeyType::ED25519, "hello.test0");
+
+    let amount = 10u128.pow(24);
+    let create_account_tx = SignedTransaction::create_account(
+        1,
+        signer0.account_id.clone(),
+        new_account_id.clone(),
+        amount,
+        new_signer.public_key.clone(),
+        &signer0,
+        *genesis_block.hash(),
+    );
+    let res = env.clients[0].process_tx(create_account_tx, false, false);
+    assert_matches!(res, ProcessTxResponse::ValidTx);
+    for i in 1..5 {
+        env.produce_block(0, i);
+    }
+
+    // add two more keys so that the account is no longer a zero balance account
+    let new_key1 = PublicKey::from_seed(KeyType::ED25519, "random1");
+    let new_key2 = PublicKey::from_seed(KeyType::ED25519, "random2");
+
+    let head = env.clients[0].chain.head().unwrap();
+    let add_key_tx = SignedTransaction::from_actions(
+        head.height * AccessKey::ACCESS_KEY_NONCE_RANGE_MULTIPLIER + 1,
+        new_account_id.clone(),
+        new_account_id.clone(),
+        &new_signer,
+        vec![
+            AddKey(AddKeyAction { public_key: new_key1, access_key: AccessKey::full_access() }),
+            AddKey(AddKeyAction { public_key: new_key2, access_key: AccessKey::full_access() }),
+        ],
+        *genesis_block.hash(),
+    );
+    let res = env.clients[0].process_tx(add_key_tx, false, false);
+    assert_matches!(res, ProcessTxResponse::ValidTx);
+    for i in 5..10 {
+        env.produce_block(0, i);
+    }
+
+    // since the account is no longer zero balance account, it cannot transfer all its tokens out
+    // and must keep some amount for storage staking
+    let transaction_costs = RuntimeConfig::test().fees;
+    let send_money_total_gas = transaction_costs.fee(ActionCosts::transfer).send_fee(false)
+        + transaction_costs.fee(ActionCosts::new_action_receipt).send_fee(false)
+        + transaction_costs.fee(ActionCosts::transfer).exec_fee()
+        + transaction_costs.fee(ActionCosts::new_action_receipt).exec_fee();
+    let total_cost = send_money_total_gas as Balance * min_gas_price;
+
+    let head = env.clients[0].chain.head().unwrap();
+    let send_money_tx = SignedTransaction::send_money(
+        head.height * AccessKey::ACCESS_KEY_NONCE_RANGE_MULTIPLIER + 1,
+        new_account_id.clone(),
+        signer0.account_id.clone(),
+        &new_signer,
+        amount - total_cost,
+        *genesis_block.hash(),
+    );
+    let res = env.clients[0].process_tx(send_money_tx, false, false);
+    assert_matches!(res, ProcessTxResponse::InvalidTx(InvalidTxError::LackBalanceForState { .. }));
```

Comment:
one thing I'd have added here: that if you removed one of the keys now, then you should be able to transfer everythign out (as the account would become a zero-balance, right?)